src/picongpu/include/simulationControl/MySimulation.hpp

```diff
@@ -462,7 +462,7 @@ class MySimulation : public SimulationHelper<simDim>
         auto fieldE = dc.get< FieldE >( FieldE::getName(), true );
         auto fieldB = dc.get< FieldB >( FieldB::getName(), true );
 
-        if( step != 0 )
+        if( this->restartRequested )
         {
             namespace nvfct = PMacc::nvidia::functors;
 
```

Comment:
on the other hand, our logic now runs `runOneStep(currentStep);` again for the *restart step*! so the above `step = this->restartStep;` should actually keep the `+1u`!